Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp

```diff
@@ -913,3 +916,407 @@ int CLuaEngineDefs::EngineGetVisibleTextureNames(lua_State* luaVM)
+                case SURFACE_PROPERTY_FRACTIONEFFECT:
+                    lua_pushnumber(luaVM, pSurface->m_frictionEffect);
+                    return 1;
+                    break;
+                case SURFACE_PROPERTY_TYREGRIP:
+                    lua_pushnumber(luaVM, pSurface->m_tyreGrip);
+                    return 1;
+                    break;
+                case SURFACE_PROPERTY_WETGRIP:
+                    lua_pushnumber(luaVM, pSurface->m_wetGrip);
+                    return 1;
+                    break;
+                case SURFACE_PROPERTY_ADHESIONGROUP:
+                    lua_pushstring(luaVM, cSurfaceAdhesionGroup[pSurface->m_adhesionGroup]);
+                    return 1;
+                    break;
+                case SURFACE_PROPERTY_CLIMBING:
+                    lua_pushboolean(luaVM, pSurface->getFlagEnabled(1, 9));
+                    return 1;
+                    break;
+                case SURFACE_PROPERTY_WHEELEFFECT:
+                    for (char cFlag = SURFACE_WHEEL_EFFECT_DISABLED; cFlag <= SURFACE_WHEEL_EFFECT_DUST; cFlag++)
+                    {
+                        if (pSurface->getFlagEnabled(1, cFlag))
+                        {
+                            lua_pushstring(luaVM, cSurfaceWheelEffect[cFlag - 1]);
+                            return 1;
+                        }
+                    }
+
+                    lua_pushstring(luaVM, "disabled");
+                    return 1;
+                    break;
+                case SURFACE_PROPERTY_STAIRS:
+                    lua_pushboolean(luaVM, pSurface->getFlagEnabled(0, 20));
+                    return 1;
+                    break;
+                case SURFACE_PROPERTY_ROUGHNESS:
+                    lua_pushnumber(luaVM, pSurface->m_roughness);
+                    return 1;
+                    break;
+            }
+        }
+        else
+            argStream.SetCustomError("Expected valid surface ID ( 0 - 179 ) at argument 1");
+    }
+    if (argStream.HasErrors())
+        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
+
+    lua_pushboolean(luaVM, true);
+    return 1;
+}
+
+int CLuaEngineDefs::EngineResetSurfaceProperties(lua_State* luaVM)
+{
+    CScriptArgReader argStream(luaVM);
+    if (argStream.NextIsNumber())
+    {
+        short sSurfaceID;
+        argStream.ReadNumber(sSurfaceID);
+        if (!argStream.HasErrors())
+        {
+            if (sSurfaceID >= EColSurfaceValue::DEFAULT && sSurfaceID <= EColSurfaceValue::SIZE)
+            {
+                lua_pushboolean(luaVM, CStaticFunctionDefinitions::ResetSurfaceInfo(sSurfaceID));
+                return 1;
+            }
+        }
+    }
+    else
+    {
+        lua_pushboolean(luaVM, CStaticFunctionDefinitions::ResetAllSurfaceInfo());
+        return 1;
+    }
+    if (argStream.HasErrors())
+        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
+    lua_pushboolean(luaVM, false);
```

Comment:
Use `lua_pushnil(luaVM);` because the code above can only fail due to invalid input.